scipy/stats/_distn_infrastructure.py

```diff
@@ -35,6 +35,8 @@
 import numpy as np
 
 from ._constants import _XMAX
+from typing import Optional
+import math
```

Comment:
Remove `math` in favor of NumPy.

Also, I am not sure about the type hints in this file. @mdhaber what do you think? We should maybe add a `_distn_infrastructure.pyi`. There seem to be more `pyi` than the other way around and it would be better to be consistent here.

Reply:
I don't know anything about type hints. I don't think it is policy to require them; they're just nice to have. So if the author is willing to contribute them, great, but if not, I don't think it should block.

Reply:
In fact, `values` should be `ArrayLike` in `numpy>1.20`. Do you have a compatibility shim for using `ArrayLike` when available and falling back to `np.ndarray` if not?

Reply:
IIRC you can just add them. Use `from future import annotations` at the top of the file, that will delay evaluation of the annotations, so if the numpy feature is not available nothing goes wrong.

We discussed this for `scipy.stats._qmc` - that has a stub file rather than inline annotations, but I believe it'll work either way.

Reply:
Yes, but in that case it won't typecheck for `numpy<1.20`, right?

I was thinking of something like:
```python
try:
    from numpy.typing import ArrayLike
except ImportError:
    ArrayLike = Any
```
This could be in a utils file for using it whenever necessary. Then you could typecheck code that uses previous numpy versions, although it will not detect all possible errors.

If nobody is going to ever typecheck the code with numpy versions prior to 1.20 your solution is fine though. If that is the case let me know and I will add that change.

Reply:
IMO it would be better with a stub file as I would not want to add yet another wait to handle type hints.

Reply:
Long-term we want everything inline, that's definitely better. But it currently escapes my memory what the blocker was in `stats._qmc`.

> If nobody is going to ever typecheck the code with numpy versions prior to 1.20 your solution is fine though. If that is the case let me know and I will add that change.

NumPy 1.20.0 was the first version to have support for type annotations and shipping with a `py.typed` file. I think it's fine to say that's our minimum bar, there's very little that will be useful for users if numpy doesn't type check.

Reply:
@rgommers I found the conversation 😅 https://github.com/scipy/scipy/pull/10844#pullrequestreview-559980027

You tried playing with `autodoc_typehints = 'none'` but the warnings I had for `WARNING: py:class reference target not found: 'float'` was still present.

Reply:
Changed type to `ArrayLike` and added `from __future__ import annotations`.

Reply:
Ah yes indeed, thanks @tupui. The final comment is https://github.com/scipy/scipy/pull/10844#issuecomment-752955763, and the tracking issue is gh-11749. Inline annotations are indeed not working right now.